integration_tests/examples/demo/utils/inputs/rtp.rs

```diff
@@ -0,0 +1,235 @@
+use anyhow::{anyhow, Result};
+use inquire::Select;
+use integration_tests::{
+    ffmpeg::start_ffmpeg_send,
+    gstreamer::{start_gst_send_tcp, start_gst_send_udp},
+};
+use serde_json::json;
+use strum::{Display, EnumIter, IntoEnumIterator};
+use tracing::error;
+
+use crate::utils::{
+    get_free_port,
+    inputs::{input_name, AudioDecoder, InputHandler, VideoDecoder},
+    TransportProtocol, IP,
+};
+
+#[derive(Debug, Display, EnumIter, Clone)]
+pub enum RtpRegisterOptions {
+    #[strum(to_string = "Add video stream")]
+    AddVideoStream,
+
+    #[strum(to_string = "Add audio stream")]
+    AddAudioStream,
+
+    #[strum(to_string = "Remove video stream")]
+    RemoveVideoStream,
+
+    #[strum(to_string = "Remove audio stream")]
+    RemoveAudioStream,
+
+    #[strum(to_string = "Set transport protocol")]
+    SetTransportProtocol,
+
+    #[strum(to_string = "Done")]
+    Done,
+}
+
+#[derive(Debug)]
+pub struct RtpInput {
+    name: String,
+    port: u16,
+    video: Option<RtpInputVideoOptions>,
+    audio: Option<RtpInputAudioOptions>,
+    transport_protocol: TransportProtocol,
+}
+
+impl RtpInput {
+    pub fn setup() -> Result<Self> {
+        let mut rtp_input = Self {
+            name: input_name(),
+
+            // TODO: (@jbrs) Make it possible for user
+            // to set their own port
+            port: get_free_port(),
+            video: None,
+            audio: None,
+            transport_protocol: TransportProtocol::Udp,
+        };
+
+        let options = RtpRegisterOptions::iter().collect::<Vec<_>>();
+        loop {
+            let action = Select::new("What to do?", options.clone()).prompt()?;
+
+            match action {
+                RtpRegisterOptions::AddVideoStream => rtp_input.setup_video()?,
+                RtpRegisterOptions::AddAudioStream => rtp_input.setup_audio()?,
+                RtpRegisterOptions::RemoveVideoStream => {
+                    rtp_input.video = None;
+                    println!("Video stream removed!");
+                }
+                RtpRegisterOptions::RemoveAudioStream => {
+                    rtp_input.audio = None;
+                    println!("Audio stream removed!");
+                }
+                RtpRegisterOptions::SetTransportProtocol => rtp_input.setup_transport_protocol()?,
+                RtpRegisterOptions::Done => {
+                    if rtp_input.video.is_none() && rtp_input.audio.is_none() {
+                        error!("At least one of \"video\" and \"audio\" has to be defined.");
+                    } else {
+                        break;
+                    }
+                }
+            }
+        }
+
+        Ok(rtp_input)
+    }
+
+    fn setup_video(&mut self) -> Result<()> {
```

Comment:
I'm not a fan off this "builder like" pattern, mixinig input object that represents state and process of providing information will not end well in my opinion.